### PR TITLE
Fix accessing trimmed paths

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Paths with trailing spaces or periods are normally trimmed by Windows. They can, however, be created using `\\?\` or be exposed via mapping of remote file systems. This change fixes the ability to work with returned *Info classes on files of this type.

This change also cleans up the Info classes:

- removes non-relevant comments
- removes "display" paths (limited trust holdover)

Also remove dead PathPair class.

### CODE REVIEW

File: src/System.IO.FileSystem/tests/Enumeration/TrimmedPaths.netcoreapp.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            File.Create(@"\\?\" + Path.Combine(directory.FullName, "Trailing period.")).Dispose();
+
+            var files = directory.GetFiles();
+            Assert.Equal(2, files.Count());
+            FSAssert.EqualWhenOrdered(new string[] { "Trailing space ", "Trailing period." }, files.Select(f => f.Name));
+
+            var paths = Directory.GetFiles(directory.FullName);
+            Assert.Equal(2, paths.Count());
+            FSAssert.EqualWhenOrdered(new string[] { "Trailing space ", "Trailing period." }, paths.Select(p => Path.GetFileName(p)));
+        }
```
Comment (pjanotti) [inline src/System.IO.FileSystem/tests/Enumeration/TrimmedPaths.netcoreapp.cs]: >} [](start = 8, length = 1)

Q.: just to confirm: is the practice to leave such files on the disk? I know that they don't have data but I want to confirm the practice here.

Comment (pjanotti) [reply src/System.IO.FileSystem/tests/Enumeration/TrimmedPaths.netcoreapp.cs]: Hum reading the test below it seems that the first is redundant and it actually deletes the files...

---
In reply to: [166747719](https://github.com/dotnet/corefx/pull/26918#discussion_r166747719) [](ancestors = 166747719)

File: src/Common/src/System/IO/PathInternal.Windows.cs [hunk 2]
```diff
@@ -81,13 +81,25 @@ internal static bool IsValidDriveChar(char value)
             return ((value >= 'A' && value <= 'Z') || (value >= 'a' && value <= 'z'));
         }
 
+        private static bool EndsWithPeriodOrSpace(string path)
+        {
+            if (string.IsNullOrEmpty(path))
```
Comment (pjanotti) [inline src/Common/src/System/IO/PathInternal.Windows.cs]: >string.IsNullOrEmpty(path) [](start = 16, length = 26)

This allows a completely blank file name and them the user can create " " and "    " hopefully nobody will do that, but yes, Windows does support it :smile:

Comment (JeremyKuhne) [reply src/Common/src/System/IO/PathInternal.Windows.cs]: Yeah, not recommended. We still make it difficult to create them by accident, but at least now we won't fall over.

File: src/System.IO.FileSystem/tests/Enumeration/TrimmedPaths.netcoreapp.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.IO.FileSystem/tests/Enumeration/TrimmedPaths.netcoreapp.cs]: The base class will clean up. It does overlap with the other test, but this discrete one will help navigate to failures a bit quicker.

**Pull request merged**